src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.targets

```diff
@@ -37,6 +37,10 @@
     <IlcSetupPropertiesDependsOn>ImportRuntimeIlcPackageTarget</IlcSetupPropertiesDependsOn>
     <IlcDynamicBuildPropertyDependencies>SetupProperties</IlcDynamicBuildPropertyDependencies>
 
+    <!-- NativeAOT apps do not require deps.json and runtimeconfig.json files -->
+    <GenerateDependencyFile>false</GenerateDependencyFile>
+    <GenerateRuntimeConfigurationFiles>false</GenerateRuntimeConfigurationFiles>
```

Comment:
I thought NativeAOT was a publish step. Won't this also affect build and make the build output incomplete?

Reply:
Moved the properties under a target that will be invoked only during publish.